homeassistant/components/systemmonitor/coordinator.py

```diff
@@ -2,27 +2,55 @@
 
 from __future__ import annotations
 
-from abc import abstractmethod
+from dataclasses import dataclass
 from datetime import datetime
 import logging
 import os
-from typing import NamedTuple, TypeVar
+from typing import Any, NamedTuple
 
 from psutil import Process
 from psutil._common import sdiskusage, shwtemp, snetio, snicaddr, sswap
 import psutil_home_assistant as ha_psutil
 
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_component import DEFAULT_SCAN_INTERVAL
-from homeassistant.helpers.update_coordinator import (
-    TimestampDataUpdateCoordinator,
-    UpdateFailed,
-)
+from homeassistant.helpers.update_coordinator import TimestampDataUpdateCoordinator
 from homeassistant.util import dt as dt_util
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, kw_only=True)
```

Comment:
```suggestion
@dataclass(frozen=True, kw_only=True, slots=True)
```


untested, but likely works fine for this case